backend/src/users/users.service.ts

```diff
@@ -23,4 +23,175 @@ export class UsersService {
+        peerId,
+        type: 'OUTGOING',
+      },
+    });
+    const createRelation2 = await this.prisma.relationship.create({
+      data: {
+        userId: peerId,
+        peerId: userId,
+        type: 'INCOMING',
+      },
+    });
+
+    return [createRelation1, createRelation2];
+  }
+
+  async acceptRequest(
+    userId: string,
+    peerId: string
+  ): Promise<[Relationship, Relationship]> {
+    const updateRelation1 = await this.prisma.relationship.update({
+      where: {
+        userId_peerId: {
+          userId,
+          peerId,
+        },
+      },
+      data: {
+        type: 'FRIEND',
+      },
+    });
+    const updateRelation2 = await this.prisma.relationship.update({
+      where: {
+        userId_peerId: {
+          userId: peerId,
+          peerId: userId,
+        },
+      },
+      data: {
+        type: 'FRIEND',
+      },
+    });
+
+    return [updateRelation1, updateRelation2];
+  }
+
+  async addFriend(
+    userId: string,
+    peerId: string
+  ): Promise<[Relationship, Relationship]> {
+    if (userId === peerId) {
+      throw new BadRequestException("can't create relation with myself");
+    }
+    const relation = await this.prisma.relationship.findUnique({
+      where: {
+        userId_peerId: {
+          userId,
+          peerId,
+        },
+      },
+    });
+
+    switch (relation?.type) {
+      case undefined:
+        return await this.request(userId, peerId);
+      case 'FRIEND':
+        throw new BadRequestException('Already friend with target');
+      case 'INCOMING':
+        return await this.acceptRequest(userId, peerId);
+      case 'OUTGOING':
+        throw new BadRequestException('Already send friend-request for target');
+      default:
+        throw new BadRequestException('Unexpected error in addFriend');
+    }
+  }
+
+  async deleteRelation(
+    userId: string,
+    peerId: string
+  ): Promise<[Relationship, Relationship]> {
+    const deleteRelation1 = await this.prisma.relationship.delete({
+      where: {
+        userId_peerId: {
+          userId,
+          peerId,
+        },
+      },
+    });
+    const _deleteRelation2 = await this.prisma.relationship.delete({
+      where: {
+        userId_peerId: {
+          userId: peerId,
+          peerId: userId,
+        },
+      },
+    });
+
+    return [deleteRelation1, _deleteRelation2];
+  }
+
+  async deleteFriend(
+    userId: string,
+    peerId: string
+  ): Promise<[Relationship, Relationship]> {
+    if (userId === peerId) {
+      throw new BadRequestException("can't create relation with myself");
+    }
+    const relation = await this.prisma.relationship.findUnique({
+      where: {
+        userId_peerId: {
+          userId,
+          peerId,
+        },
+      },
+    });
+
+    switch (relation?.type) {
+      case undefined:
+        throw new BadRequestException(
+          'You are not friend with target and do not send to friend-request'
+        );
+      case 'FRIEND':
```

Comment:
```suggestion
      case 'FRIEND':
      case 'INCOMING':
      case 'OUTGOING':
            return await ~
```
でいいと思いました。なんならdefault でも。